x/sequencer/keeper/msg_server_increase_bond.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension/v3/x/sequencer/types"
+)
+
+// IncreaseBond implements types.MsgServer.
+func (k msgServer) IncreaseBond(goCtx context.Context, msg *types.MsgIncreaseBond) (*types.MsgIncreaseBondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sequencer, err := k.bondUpdateAllowed(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// transfer the bond from the sequencer to the module account
+	seqAcc := sdk.MustAccAddressFromBech32(msg.Creator)
+	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, seqAcc, types.ModuleName, sdk.NewCoins(msg.AddAmount))
+	if err != nil {
+		return nil, err
+	}
+
+	// update the sequencers bond amount in state
+	sequencer.Tokens = sequencer.Tokens.Add(msg.AddAmount)
+	k.UpdateSequencer(ctx, sequencer, sequencer.Status)
+
+	// emit a typed event which includes the added amount and the active bond amount
+	err = ctx.EventManager().EmitTypedEvent(
+		&types.EventIncreasedBond{
+			Sequencer:   msg.Creator,
+			Bond:        sequencer.Tokens,
+			AddedAmount: msg.AddAmount,
+		},
+	)
+
+	return &types.MsgIncreaseBondResponse{}, err
+}
+
+func (k msgServer) bondUpdateAllowed(ctx sdk.Context, msg *types.MsgIncreaseBond) (types.Sequencer, error) {
+	// check if the sequencer already exists
+	sequencer, found := k.GetSequencer(ctx, msg.Creator)
+	if !found {
+		return types.Sequencer{}, types.ErrUnknownSequencer
+	}
+
+	// check if the sequencer is bonded
+	if !sequencer.IsBonded() {
```

Comment:
@spoo-bar If it unbonded, I think u should allow it to bond again
check for min bond requirement

Reply:
@mtsitrin I think thats outside the scope of the current feature requirements. And expected behavior is to use the re-registration flow in case they unbond

Ref: https://faulttolerance-group.slack.com/archives/C07BRL21528/p1722435907634139